### PR TITLE
Fix cromwell restart

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 1]
```diff
@@ -489,9 +471,8 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       val updatedData = data.addPersisting(callKey, ExecutionStatus.Aborted)
       stay() using updatedData
     case Event(message: CallStartMessage, data) if !data.isPending(message.callKey) =>
-      val updatedData = message.handleStatusPersist(this, data)
       startCallWithMessage(message)
-      stay() using updatedData
+      stay() using data
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: you can underscore `data` in the pattern match and omit `using data` here

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 2]
```diff
@@ -87,36 +87,18 @@ object WorkflowActor {
...
-  final case class RestartCall(override val callKey: CallKey, override val startMode: CallActor.StartMode) extends CallStartMessage {
-    override def handleStatusPersist(actor: WorkflowActor, data: WorkflowData)(implicit logger: WorkflowLogger): WorkflowData = {
-      actor.executionStore += callKey -> ExecutionStatus.Starting
-      // The assumption for restart is that the call will have been found in state Running.  Explicitly
-      // re-writing the Starting status causes the new start date to be persisted, which is desirable to
-      // provide a realistic elapsed execution time.
-      actor.persistStatus(callKey, ExecutionStatus.Starting)
-      data.addPersisting(callKey, ExecutionStatus.Starting)
-    }
-  }
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: good to see lots of red lines in `WorkflowActor` :smile:

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala (hunk 1 above)
Comment (Horneth) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: :+1:

File: src/main/scala/cromwell/engine/backend/jes/JesBackend.scala [hunk 3]
```diff
@@ -75,6 +75,11 @@ object JesBackend {
       eis.exists(ei => ei.key == JesBackend.InfoKeys.JesRunId && ei.value.isDefined)
   }
 
+  private val IsResetable: (Execution, Seq[ExecutionInfo]) => Boolean = (e: Execution, eis: Seq[ExecutionInfo]) => {
+    e.status.toExecutionStatus == ExecutionStatus.Running &&
+      eis.exists(ei => ei.key == JesBackend.InfoKeys.JesRunId && ei.value.isEmpty)
+  }
+
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: Should this perhaps be named `IsTransient`?

File: src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala [hunk 4]
```diff
@@ -726,12 +726,12 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
...
   }
 
-  override def resetNonResumableExecutions(workflowId: WorkflowId, isResumable: (Execution, Seq[ExecutionInfo]) => Boolean)(implicit ec: ExecutionContext): Future[Unit] = {
+  override def resetTransientExecutions(workflowId: WorkflowId, isResetable: (Execution, Seq[ExecutionInfo]) => Boolean)(implicit ec: ExecutionContext): Future[Unit] = {
     val action = for {
       tuples <- dataAccess.executionsAndExecutionInfosByWorkflowId(workflowId.toString).result
       executionsAndInfos = tuples groupBy { _._1 } mapValues { _ map { _._2 } }
 
-      nonResumableDatabaseKeys = (executionsAndInfos filterNot Function.tupled(isResumable)).keys map { _.toKey }
+      nonResumableDatabaseKeys = (executionsAndInfos filter Function.tupled(isResetable)).keys map { _.toKey }
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: val should probably be `transientDatabaseKeys`

File: src/main/scala/cromwell/engine/db/DataAccess.scala [hunk 5]
```diff
@@ -115,7 +115,7 @@ trait DataAccess extends AutoCloseable {
 
   def updateWorkflowOptions(workflowId: WorkflowId, workflowOptionsJson: String)(implicit ec: ExecutionContext): Future[Unit]
 
-  def resetNonResumableExecutions(workflowId: WorkflowId, isResumable: (Execution, Seq[ExecutionInfo]) => Boolean)(implicit ec: ExecutionContext): Future[Unit]
+  def resetTransientExecutions(workflowId: WorkflowId, isResetable: (Execution, Seq[ExecutionInfo]) => Boolean)(implicit ec: ExecutionContext): Future[Unit]
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/db/DataAccess.scala]: `isTransient` here too

**Comment by mcovarr** [comment]
:+1: thanks for fixing this!

File: src/main/scala/cromwell/engine/backend/jes/JesBackend.scala [hunk 6]
```diff
@@ -75,6 +75,11 @@ object JesBackend {
       eis.exists(ei => ei.key == JesBackend.InfoKeys.JesRunId && ei.value.isDefined)
   }
 
+  private val IsTransient: (Execution, Seq[ExecutionInfo]) => Boolean = (e: Execution, eis: Seq[ExecutionInfo]) => {
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: Could you add a mini-comment above this to say what `transient` means in this context? It seems to be closer to `AlreadyRunning`

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala (hunk 2 above)
Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: ![Better Red Than Dead] (http://ec-cdn-assets.stitcher.com/feedimagesplain328/44417.jpg)

**Comment by cjllanwarne** [comment]
:+1:

**Pull request merged**